From 8f87585e6bf27f65196504515417f32aa4ccc764 Mon Sep 17 00:00:00 2001
From: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Date: Tue, 19 Jul 2022 18:10:00 +0200
Subject: [PATCH 373/529] ARM: dts: stm32: adopt wakeup pin for stm32g0 typec
 in stm32mp135f-dk

USB Type-c controller (stm32g0) has an interrupt pin wired to a
STM32MP13 PWR wakeup pin on DK board. It can be used as a wakeup source,
to wakeup the system from standby mode, when changes are identified on the
Type-c connector.
All is configured in OPTEE, adopt it.

Change-Id: I23380dccb337665e06edc1c8d14caf58549efbc9
Signed-off-by: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/280697
---
 arch/arm/boot/dts/stm32mp135f-dk.dts | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

--- a/arch/arm/boot/dts/stm32mp135f-dk.dts
+++ b/arch/arm/boot/dts/stm32mp135f-dk.dts
@@ -229,13 +229,10 @@
 	typec@53 {
 		compatible = "st,stm32g0-typec";
 		reg = <0x53>;
-		/* Alert pin on PI2 */
-		interrupts = <2 IRQ_TYPE_EDGE_FALLING>;
-		interrupt-parent = <&gpioi>;
-		/* Internal pull-up on PI2 */
-		pinctrl-names = "default";
-		pinctrl-0 = <&stm32g0_intn_pins_a>;
+		/* Alert pin on PI2 (PWR wakeup pin), managed by optee */
+		interrupts-extended = <&optee 1>;
 		firmware-name = "stm32g0-ucsi.mp135f-dk.fw";
+		wakeup-source;
 		connector {
 			compatible = "usb-c-connector";
 			label = "USB-C";
